### ISSUE TITLE
Deploy GH wiki Rust proxy

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi_documentation#138]
https://github.com/nelsonjchen/github-wiki-see-rs

need a box where we can drop a docker and park rusefi subdomain

**Comment by rusefillc** [comment]
@chuckwagoncomputing would you like root access to

![image](https://user-images.githubusercontent.com/48498823/118544119-32dcc680-b723-11eb-94a7-a0346cea9699.png)

**Comment by rusefillc** [comment]
https://github.com/djacquel/WikiToJekyll is an alternative

**Comment by chuckwagoncomputing** [comment]
@rusefillc I think we should do the Github Pages option. If you're on board with that, you should create a repo and set up a .rusefi.com subdomain to point to it.

https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site

**Comment by rusefillc** [comment]
https://github.com/rusefi/rusefi.github.io created
GH is aware of custom domain now https://github.com/rusefi/rusefi.github.io/blob/main/CNAME

![image](https://user-images.githubusercontent.com/48498823/118591221-d359d780-b771-11eb-87a4-2121b636ed1d.png)

**Comment by rusefillc** [comment]
Now I am confused why does http://rusefi.github.io redirect me to http://wiki.rusefi.com?

**Comment by rusefillc** [comment]
![image](https://user-images.githubusercontent.com/48498823/118592400-dfdf2f80-b773-11eb-88cc-387bbbbf659c.png)

**Comment by rusefillc** [comment]
certificate drama next?

![image](https://user-images.githubusercontent.com/48498823/118592433-ee2d4b80-b773-11eb-9d15-be1ef7f6e7bf.png)

**Comment by chuckwagoncomputing** [comment]
https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/troubleshooting-custom-domains-and-github-pages#https-errors

**Comment by rusefillc** [comment]
is TL,DR to wait an hour?
It has been a few and still

![image](https://user-images.githubusercontent.com/48498823/118686518-0f248980-b7d2-11eb-9fb9-9e23aa85c1e7.png)

**Comment by chuckwagoncomputing** [comment]
Try removing and re-adding the custom domain in the repo settings.

**Comment by rusefillc** [comment]
![image](https://user-images.githubusercontent.com/48498823/118691489-21ed8d00-b7d7-11eb-8f7e-ac8737dcb60e.png)


will give it a couple of hours and re-add

**Comment by rusefillc** [comment]
IT HAS WORKED?!

![image](https://user-images.githubusercontent.com/48498823/118727897-86bddd00-b801-11eb-992c-fce25bbd4c28.png)

**Comment by rusefillc** [comment]
![image](https://user-images.githubusercontent.com/48498823/118728181-be2c8980-b801-11eb-8fa2-3d350b805187.png)

**Comment by rusefillc** [comment]
Need help with all this :(

https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll
And the plugin and config from https://github.com/djacquel/WikiToJekyll

**Comment by chuckwagoncomputing** [comment]
So it turns out that Github Pages only supports a subset of Jekyll plugins, so the WikiToJekyll plugin doesn't work. We can generate it with a Github Action though.

**Comment by chuckwagoncomputing** [comment]
So basically, at this point there's no purpose to using Github Pages. In fact we'll be ahead with the configurability of our own server.
New Plan: Use a Github Action to generate static site files and FTP them to the server.

As to which static site generator to use, I'm not sure yet. Either I don't understand Jekyll infrastructure, or it's deeply crippled for non-github-pages use.

**Comment by chuckwagoncomputing** [comment]
Status update:
My next idea was Readthedocs. Readthedocs has a "Github Import" feature, but it doesn't seem to know how to handle markdown properly.
So I decided to try Sphinx, which is what Readthedocs runs on. There is a markdown parser extension for it. This works reasonably well, but has a few issues I haven't figured out how to handle.
1. If you add a file to the table of contents, all of the headers (lines starting with # in markdown) in that file are listed. For example, the Home page appears in the index as 4 items: 
    Click on the image below for the latest builds and downloads!
    Features
    Get Started!
    rusEFI in action!
The Sphinx index system has a few options that I thought might fix this, but it seems like this is how things are getting passed from the parser, and it doesn't work.

2. Files that don't have a header aren't listed in the index. Github gets its titles from the .md file name, but this doesn't seem possible with Sphinx. However, adding headers to files would be an improvement in Github as well, so I don't see this as a big problem. If we need to modify the markdown parser to fix 1 though, we could probably fix this too.

3. I noticed that with sphinx-rtd-theme, table cells don't wrap, and on the Hardware table, you have to scroll horizontally to see more than a few columns, at least on my 1366x768 laptop screen. This is a downgrade from the Github wiki.

**Comment by chuckwagoncomputing** [comment]
Now I have tried using Pandocs to convert all the .md files to .rst first. This fixes problem 1, but breaks tables in a new and interesting way. Pandoc makes large cells multiple lines high, and Sphinx doesn't handle that properly.

**Comment by chuckwagoncomputing** [comment]
Now I have tried Gollum, which is what Github Wikis use behind the scenes. It has problems too.
1. Tables don't work, and I don't know why.
2. Links to pages in a directory don't work. Gollum docs say you can use --lenient-tag-lookup to make this work, but it doesn't.

Besides these, it's fairly nice.

**Comment by chuckwagoncomputing** [comment]
I have found MkDocs and it is beautiful.

**Comment by chuckwagoncomputing** [comment]
We still need to add "title" headers, but otherwise I'm happy with this. Demo at https://rusefi.chuckwagoncomputing.com/wiki/Home/

**Comment by chuckwagoncomputing** [comment]
OK, so I'm happy with how things have worked out with MkDocs, and am ready to PR it. I have one question though. Do we want to keep the Github Wiki? Both are compatible with the current state of the wiki, but if we take advantage of things MkDocs can do (Youtube embeds!) then the Github Wiki will be broken.
My vote is to drop it like a hot potato. :)

**Comment by rusefillc** [comment]
I vote for one wiki whichever is best

**Comment by chuckwagoncomputing** [comment]
Noting this here so you don't forget:
The main page of the wiki is https://rusefi.com/docs/wiki/Home/
Whatever URL you want to use will need to be redirected to that.
e.g.
https://rusefi.com/docs/wiki
https://rusefi.com/wiki
https://wiki.rusefi.com
Or as many as you like, though we should probably pick one as "official".

**Comment by rusefillc** [comment]
definitely not https://rusefi.com/docs/wiki :) I will add/fix secrets once we are closer to production

**Comment by rusefillc** [comment]
Depends on #144

**Comment by rusefillc** [comment]
Depends on #145

**Comment by rusefillc** [comment]
Depends on #146

**Comment by rusefillc** [comment]
Depends on #149

**Comment by nelsonjchen** [comment]
😅 , my terrible web app hack wasn't meant to be an instance'd thing (https://github-wiki-see.page/). I'm glad you guys are sorting that out though!

**Comment by chuckwagoncomputing** [comment]
I saw another option today that may be worth considering: Docusaurus.
https://github.com/facebook/docusaurus/issues/2725

**Comment by rusefillc** [comment]
As of today https://github.com/robots.txt does not mention wiki. it looks like this issue is no longer an issue.

We now need to clean-up wiki1 and consider hiding rusefi_documentation

**Comment by chuckwagoncomputing** [comment]
Current status is that Google indexing of Wiki3 still sucks, but Bing and DuckDuckGo indexing is great.

**Comment by mi-hol** [comment]
> Google indexing of Wiki3 still sucks

@chuckwagoncomputing  I seem to remember a hint in SiteChecker about "duplicated content" and a note to configure a corresponding hint for Google. Should I dig for this info?

**Comment by chuckwagoncomputing** [comment]
I made a sitechecker account, and these are the issues listed under Indexability:
![image](https://user-images.githubusercontent.com/1531127/212486817-6812a70f-2763-4e51-9d73-03296e14d5a2.png)
I'm not sure how big of issues these actually are. From my research, I don't think that not having a robots.txt is an issue.  
Other tags in H1 and H2 are because of how MkDocs generates the header hotlinks. That could be turned off, but it's nice to have. If turned off, hotlinks would still work, they just wouldn't be discoverable.

**Comment by chuckwagoncomputing** [comment]
Here is the Duplicate Content section
![image](https://user-images.githubusercontent.com/1531127/212487100-2ec0f407-858e-43f3-b884-d13d9617c211.png)
